docs/source/udf/profiling.rst

```diff
@@ -0,0 +1,180 @@
+.. _`udf profiling`:
+
+.. testsetup:: *
+
+    from libertem.api import Context
+
+
+Profiling UDFs
+==============
+
+To bring your UDF to production, it may be necessary to look into the performance and
+efficiency of your function. There may be some low-hanging fruit to optimize away, but
+they only really become visible once you profile your code. Just looking at code and guessing
+which parts are expensive can be misleading and cause you to spend time optimizing the wrong part
+of your code. Always measure!
+
+Profiling means instrumenting the execution of a program and finding out which parts
+of your code use how much of a given resource (for example, CPU time, or memory). 
+
+Prerequisite: inline executor
+-----------------------------
+
+By default, your UDF will be run in a multi-process or multi-threaded environment. This makes
+profiling a challenge. But as most of the time, improving the execution time of your UDF in a
+single-threaded executor also improves the respective multi-process execution time, we can run
+the UDF in a single thread to do profiling measurements.
+
+LiberTEM provides what it calls :class:`~libertem.executor.inline.InlineJobExecutor`,
+which executes your UDF in a single thread, thus allowing the use of default Python profiling
+mechanisms. 
+
+To use it, pass it to the :class:`~libertem.api.Context` on construction:
```

Comment:
This description of how to use the InlineJobExecutor is redundant with https://github.com/LiberTEM/LiberTEM/blob/master/docs/source/debugging.rst#debugging-udfs-or-other-python-code

Perhaps the part on debugging the Python part of LiberTEM can somehow be combined with profiling since they are conceptually very similar?

Reply:
Hmm, I would prefer to not merge, the `InlineJobExecutor` instructions are short enough I think. Also would not want to cross-reference too much, which would require jumping around too much to understand.

Reply:
Ok, makes sense!